src/Features/CSharp/Portable/EncapsulateField/CSharpEncapsulateFieldService.cs

```diff
@@ -140,28 +142,38 @@ private bool CanEncapsulate(FieldDeclarationSyntax field)
             return field.Parent is TypeDeclarationSyntax;
         }
 
-        protected override Tuple<string, string> GeneratePropertyAndFieldNames(IFieldSymbol field)
+        private static ImmutableArray<NamingRule> GetNamingRules(Document document, CancellationToken cancellationToken)
         {
+            return document.GetNamingRulesAsync(cancellationToken).GetAwaiter().GetResult()
+                .AddRange(DefaultNamingRules.FieledAndPropertyRules);
+        }
+
+        protected override Tuple<string, string> GeneratePropertyAndFieldNames(IFieldSymbol field, Document document, CancellationToken cancellationToken)
+        {
+            var namingRules = GetNamingRules(document, cancellationToken);
             // Special case: if the field is "new", we will preserve its original name and the new keyword.
             if (field.DeclaredAccessibility == Accessibility.Private || IsNew(field))
             {
                 // Create some capitalized version of the field name for the property
-                return Tuple.Create(field.Name, MakeUnique(GeneratePropertyName(field.Name), field.ContainingType));
+                return Tuple.Create(field.Name, MakeUnique(GeneratePropertyName(field.Name), field.ContainingType, namingRules, SymbolKind.Property, Accessibility.Public));
             }
             else
             {
                 // Generate the new property name using the rules from 695042
-                var newPropertyName = GeneratePropertyName(field.Name);
+                var basePropertyName = GeneratePropertyName(field.Name);
```

Comment:
Not sure how to handle this. Should we try to remove naming style from field name before generating a property name?

Reply:
We should do the same thing we do for #33673